### PR TITLE
:seedling: upgrade sdk-go

### PR DESCRIPTION
**Pull request by skeeey**
## Related issue(s)

Fixes #

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -36,7 +36,7 @@ require (
 	k8s.io/utils v0.0.0-20240310230437-4693a0247e57
 	open-cluster-management.io/addon-framework v0.9.1-0.20240419070222-e703fc5a2556
 	open-cluster-management.io/api v0.13.1-0.20240521030453-9d94703b9eba
-	open-cluster-management.io/sdk-go v0.13.1-0.20240520073308-f18d198a844d
+	open-cluster-management.io/sdk-go v0.13.1-0.20240605032202-2a0633ddf34e
```
Comment (morvencao) [inline go.mod:39]: @skeeey please update to latest commit: https://github.com/open-cluster-management-io/sdk-go/commit/874c2c7ac523ce73503b8322611b17510363c33f

there should no code change for your PR, just upgrade the dependency to get this fix: https://github.com/open-cluster-management-io/sdk-go/pull/52

Comment (skeeey) [reply go.mod]: done

File: pkg/work/hub/controllers/manifestworkreplicasetcontroller/manifestworkreplicaset_deploy_reconcile.go [hunk 2]
```diff
@@ -278,8 +278,8 @@ func CreateManifestWork(mwrSet *workapiv1alpha1.ManifestWorkReplicaSet, clusterN
 				ManifestWorkReplicaSetPlacementNameLabelKey:  placementRefName,
 			},
 			Annotations: map[string]string{
-				common.CloudEventsDataTypeAnnotationKey:   payload.ManifestBundleEventDataType.String(),
-				common.CloudEventsGenerationAnnotationKey: fmt.Sprintf("%d", mwrSet.Generation),
+				common.CloudEventsDataTypeAnnotationKey:        payload.ManifestBundleEventDataType.String(),
```
Comment (qiujian16) [inline pkg/work/hub/controllers/manifestworkreplicasetcontroller/manifestworkreplicaset_deploy_reconcile.go:281]: could you explain why these annotation is needed? should it be optionally set?

Comment (skeeey) [reply pkg/work/hub/controllers/manifestworkreplicasetcontroller/manifestworkreplicaset_deploy_reconcile.go]: add the comments, these are only needed by cloudevents client

Comment (skeeey) [reply pkg/work/hub/controllers/manifestworkreplicasetcontroller/manifestworkreplicaset_deploy_reconcile.go]: only add them when cloudevents client is used

Comment (skeeey) [reply pkg/work/hub/controllers/manifestworkreplicasetcontroller/manifestworkreplicaset_deploy_reconcile.go]: removed

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**